yarn-project/noir-libs/noir-aztec/src/note/note_getter_options.nr

```diff
@@ -29,50 +40,67 @@ impl Sort {
     }
 }
 
-struct NoteGetterOptions<Note, N, S, P> {
-    sort_by: [Sort; N],
+fn return_all_notes<Note, N>(notes: [Option<Note>; MAX_READ_REQUESTS_PER_CALL], _p: Field) -> [Option<Note>; MAX_READ_REQUESTS_PER_CALL] {
+    notes
+}
+
+// docs:start:NoteGetterOptions
+struct NoteGetterOptions<Note, N, FILTER_ARGS> {
+    selects: [Option<Select>; N],
+    sorts: [Option<Sort>; N],
+    limit: u32,
     offset: u32,
-    filter: fn ([Option<Note>; MAX_READ_REQUESTS_PER_CALL], P) -> [Option<Note>; S],
-    filter_args: P,
+    filter: fn ([Option<Note>; MAX_READ_REQUESTS_PER_CALL], FILTER_ARGS) -> [Option<Note>; MAX_READ_REQUESTS_PER_CALL],
+    filter_args: FILTER_ARGS,
 }
+// docs:end:NoteGetterOptions
```

Comment:
Is there an order of precedents to these? In which order are these applied? (Presumably the order in which they're applied affects the notes which are returned, in some cases).

Reply:
`filter` and `filter_args` are applied after the notes are returned from the oracle call, and before we add the final notes to `read_requests`. (Already put this in the docs.)
The rest is like db operations: select -> sort -> offset -> limit